cmd/scollector/collectors/stream.go

```diff
@@ -0,0 +1,82 @@
+package collectors
+
+import (
+	"reflect"
+	"runtime"
+	"sync"
+	"time"
+
+	"bosun.org/collect"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+type StreamCollector struct {
+	F      func() <-chan *opentsdb.MultiDataPoint
+	Enable func() bool
+	name   string
+	init   func()
+
+	// internal use
+	sync.Mutex
+	enabled bool
+}
+
+func (s *StreamCollector) Init() {
+	if s.init != nil {
+		s.init()
+	}
+}
+
+func (s *StreamCollector) Run(dpchan chan<- *opentsdb.DataPoint, quit <-chan struct{}) {
+	if s.Enable != nil {
+		go func() {
+			for {
+				next := time.After(time.Minute * 5)
+				s.Lock()
+				s.enabled = s.Enable()
```

Comment:
is Enabled really a useful concept here? The interval collector uses it to determine "Is X installed on this system" so it can avoid issuing unecessary requests and such. If you real collector is running external to the Stream Collector, all you are doing with enabled is dropping datapoints. I would rather just always be enabled on these, and make the user "opt in" by enabling it in config.